### PR TITLE
Actuallize podresources description

### PR DESCRIPTION
**Pull request by AlexeyPerevalov**
This commit updates description according to
https://github.com/kubernetes/enhancements/pull/1884

/milestone 1.20
Issue: https://github.com/kubernetes/enhancements/issues/2043

Signed-off-by: Alexey Perevalov <alexey.perevalov@huawei.com>

### CODE REVIEW

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 1]
```diff
@@ -189,10 +189,87 @@ The kubelet provides a gRPC service to enable discovery of in-use devices, and t
...
 // node resources consumed by pods and containers on the node
-service PodResourcesLister {
+service PodResources {
     rpc List(ListPodResourcesRequest) returns (ListPodResourcesResponse) {}
+    rpc GetAllocatableResources(AllocatableResourcesRequest) returns (AllocatableResourcesResponse) {}
+    rpc Watch(WatchPodResourcesRequest) returns (stream WatchPodResourcesResponse) {}
+}
+```
+
+List endpoint provides information on resources of running pods with details such as
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:201]:
```suggestion
The `List` endpoint provides information on resources of running pods, with details such as the
```

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 2]
```diff
@@ -189,10 +189,87 @@ The kubelet provides a gRPC service to enable discovery of in-use devices, and t
...
+id of exclusively allocated CPUs, device id as it was reported by device plugins and id of
+NUMA node where these devices are allocated.
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:203]:
```suggestion
the NUMA node where these devices are allocated.
```

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 3]
```diff
@@ -189,10 +189,87 @@ The kubelet provides a gRPC service to enable discovery of in-use devices, and t
...
+}
+
+```
+
+Watch intends to quickly obtain information on a newly launched pod, the information obtained is the same
+as obtained by List. By using Watch endpoint Monitoring agents do not need to call List periodically anymore.
+
+```gRPC
+
+// WatchPodResourcesRequest is the request made to the Watch PodResourcesLister service
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:260]: `PodResourcesLister` or `PodResources`?

**Review by sftim — COMMENTED** [review]
This makes sense to me. I haven't really checked the gRPC details though.
As this page is getting long, I wonder if it makes sense to move that gRPC service description within https://k8s.io/reference/ ?

(if you agree - we can do that move as a second PR against _dev-1.20_)

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md (hunk 3 above)
Comment (AlexeyPerevalov) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:260]: Good question. In both KEP (https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2043-pod-resource-concrete-assigments, https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/606-compute-device-assignment)  PodResources was used, but in implementation PodResourcesLister is used now.
Documentation should be written according to implementation.
But what to do with KEP, rename it there?

**Comment by AlexeyPerevalov** [comment]
> This makes sense to me. I haven't really checked the gRPC details though.
> As this page is getting long, I wonder if it makes sense to move that gRPC service description within https://k8s.io/reference/ ?
> 
> (if you agree - we can do that move as a second PR against _dev-1.20_)

I had the same think when edited. I thought it's reasonable to move PodResorcesLister section into new file. But what is it - https://k8s.io/reference/? Is it https://kubernetes.io/docs/reference/?

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md (hunk 3 above)
Comment (dashpole) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:260]: Yes, feel free to update the KEP as well

**Comment by AlexeyPerevalov** [comment]
> /remove-sig architecture
> /sig node
> /hold
> waiting for [kubernetes/kubernetes#93243](https://github.com/kubernetes/kubernetes/pull/93243) to merge

Not only PR 93243, but also in this PR was mentioned API from https://github.com/kubernetes/kubernetes/pull/95734.
Both should be merged.

**Comment by annajung** [comment]
/hold
pending https://github.com/kubernetes/kubernetes/pull/93243 and https://github.com/kubernetes/kubernetes/pull/95734

**Comment by sftim** [comment]
Hi @AlexeyPerevalov 

Would you have a chance to rebase this against _dev-1.20_? It's currently got a merge conflict.

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 4]
```diff
@@ -204,7 +259,7 @@ DaemonSet, `/var/lib/kubelet/pod-resources` must be mounted as a
 {{< glossary_tooltip term_id="volume" >}} in the plugin's
 [PodSpec](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podspec-v1-core).
 
-Support for the "PodResources service" requires `KubeletPodResources` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to be enabled. It is enabled by default starting with Kubernetes 1.15.
+Support for the "PodResourcesLister service" requires `KubeletPodResources` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to be enabled. It is enabled by default starting with Kubernetes 1.15.
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:262]: nit: You can remove the double quotes around, PodResourceLister service ...
and possibly format the name of the service with single back quotes,
`PodResourcesLister`?

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 5]
```diff
@@ -193,9 +193,64 @@ for these devices:
 // node resources consumed by pods and containers on the node
 service PodResourcesLister {
     rpc List(ListPodResourcesRequest) returns (ListPodResourcesResponse) {}
+    rpc GetAllocatableResources(AllocatableResourcesRequest) returns (AllocatableResourcesResponse) {}
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:196]: This is an API offered by the kubelet? gRPC only?

Comment (tengqm) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:196]: Yes.

**Comment by irvifa** [comment]
/hold
pending https://github.com/kubernetes/kubernetes/pull/95734

**Comment by irvifa** [comment]
Hi @AlexeyPerevalov Since the code freeze is on effect, wondering if folks raise exception for this since one of the PR is not merged yet

**Comment by annajung** [comment]
Note:

Exception filed (pending approval)

**Comment by annajung** [comment]
Exception request has been denied for this enhancement. 

/milestone clear
/close

**Pull request closed without merging**